shell.nix

```diff
@@ -1,159 +1,159 @@
-      # all tests
-      alias t="ut && ft"
-
-      echo "adding ${builtins.concatStringsSep ":" binDirs} to \$PATH to make running built binaries more natural"
-      export PATH=$PATH:${builtins.concatStringsSep ":" binDirs};
-
-      alias a c m c_fast cm acm acm_nw acm_fast ut ft t
+  binDirs = [ "$PWD/build/src" "$PWD/build/src/qt" ];
+  jobs = if (strings.hasSuffix "linux" builtins.currentSystem) then
+    "$(($(nproc)-${toString spareCores}))"
+  else if (strings.hasSuffix "darwin" builtins.currentSystem) then
+    "$(($(sysctl -n hw.physicalcpu)-${toString spareCores}))"
+  else
+    "6";
+
+  libmultiprocess = pkgs.stdenv.mkDerivation {
+    name = "libmultiprocess";
+    src = pkgs.fetchFromGitHub {
+      owner = "bitcoin-core";
+      repo = "libmultiprocess";
+      rev = "f35df6bdc536b068597559d4ab470dab9cff7cfc";
+      sha256 = "sha256-1gg6MAql70JUXOdaP0A9Lmny5y6EFw55bJrzUvYRzMQ=";
+    };
+
+    nativeBuildInputs = [ pkgs.cmake pkgs.pkg-config ];
+    buildInputs = [ pkgs.capnproto ];
+    cmakeFlags =
+      [ "-DCMAKE_INSTALL_LIBDIR=lib" "-DCMAKE_INSTALL_INCLUDEDIR=include" ];
+
+    # Optional test check
+    doCheck = true;
+    checkPhase = ''
+      make check
+    '';
+
+    # Make sure pkg-config and cmake files are properly installed
+    postInstall = ''
+      mkdir -p $out/lib/pkgconfig
     '';
+
+    # Makes sure that the pkg-config and cmake files can find dependencies
+    setupHook = pkgs.writeText "setup-hook.sh" ''
+      export PKG_CONFIG_PATH="''${PKG_CONFIG_PATH:+$PKG_CONFIG_PATH:}${
+        placeholder "out"
+      }/lib/pkgconfig"
+      export CMAKE_PREFIX_PATH="''${CMAKE_PREFIX_PATH:+$CMAKE_PREFIX_PATH:}${
+        placeholder "out"
+      }"
+    '';
+
+    meta = with pkgs.lib; {
+      description = "Multi-process library";
+      homepage = "https://github.com/bitcoin-core/libmultiprocess";
+      license = licenses.mit;
+    };
+  };
+in pkgs.mkShell {
+  nativeBuildInputs = with pkgs; [
+    # Essential build tools
+    boost
+    ccache
+    clang-tools_19
+    clang_19
+    cmake
+    gcc14
+    libevent
+    pkg-config
+    sqlite
+
+    # Optional build dependencies
+    capnproto
+    db4
+    libmultiprocess
+    qrencode
+    zeromq
+
+    # Tests
+    hexdump
+
+    # Depends
+    byacc
+
+    # Functional tests & linting
+    python310
+    python310Packages.autopep8
+    python310Packages.flake8
+    python310Packages.lief
+    python310Packages.mypy
+    python310Packages.pyzmq
+    python310Packages.requests
+
+    # Benchmarking
+    python310Packages.pyperf
+
+    # Debugging
+    gdb
+
+    # Tracing
+    libsystemtap
+    linuxPackages.bcc
+    linuxPackages.bpftrace
+
+    # Bitcoin-qt
+    qt5.qtbase
+    # required for bitcoin-qt for "LRELEASE" etc
+    qt5.qttools
+  ];
+
+  # Modifies the Nix clang++ wrapper to avoid warning:
+  # "_FORTIFY_SOURCE requires compiling with optimization (-O)"
+  hardeningDisable = if withDebug then [ "all" ] else [ ];
+
+  # TODO: Is it?
```

Comment:
I never tested this out...

Reply:
I can do

Reply:
It appears to fine for me:

```bash
-- Found Boost: /nix/store/cqnj4iwzabn5crvvcigx5cjd2jif7n5y-boost-1.81.0-dev/include (found suitable version "1.81.0", minimum required is "1.73.0")
```

Reply:
Same for me. Feel free to drop it.

Reply:
Dropped and squashed the CI commits together.